### PR TITLE
Rename vagrant to local

### PR DESCRIPTION
**Pull request by mvladev**
This will allow to change local implementations to support other providers than Vagrant.

Added defaulting for `.spec.cloud.local.networks`

### CODE REVIEW

**Comment by mvladev** [comment]
I don't know why Github complaints about conflicts in `.gitignore`

```bash
$ git merge rename-vagrant-to-local-provider --no-commit --no-ff
.....
.....
Automatic merge went well; stopped before committing as requested
```

File: docs/deployment/kubernetes-minikube.md [hunk 1]
```diff
@@ -26,23 +26,23 @@ $ kubectl create clusterrolebinding add-on-cluster-admin --clusterrole=cluster-a
...
 $ helm init
 
-# Deploy Gardener with vagrant specific configuration
+# Deploy Gardener with local specific configuration
 $ helm install charts/gardener \
   --name gardener \
   --namespace garden \
   --values=charts/gardener/values.yaml \
-  --values=charts/gardener/vagrant-local-values.yaml
+  --values=charts/gardener/local-local-values.yaml
```
Comment (rfranzke) [inline docs/deployment/kubernetes-minikube.md]: local-local-values -> local-values

File: minikube.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+minikube start \
+  --extra-config=controller-manager.ClusterSigningCertFile="/var/lib/localkube/certs/ca.crt" \
+  --extra-config=controller-manager.ClusterSigningKeyFile="/var/lib/localkube/certs/ca.key" \
+  --extra-config=apiserver.Admission.PluginNames=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota \
+  --kubernetes-version=v1.9.0 \
+  --extra-config=apiserver.Authorization.Mode=RBAC
```
Comment (rfranzke) [inline minikube.sh]: How is this file related? It's not mentioned anywhere? Accidentally committed?

**Review by rfranzke — COMMENTED** [review]
Two comments, other than that it looks good so far.
You should rebase on `gardener/master` to resolve the merge conflicts.

**Comment by mvladev** [comment]
@rfranzke removed accidentally committed file and double `local-local-values.yaml`

**Comment by rfranzke** [comment]
Hey Martin, could you please rebase once again?

**Comment by mvladev** [comment]
@rfranzke rebased

**Review by rfranzke — APPROVED** [review]
